packages/editor/src/components/copy-handler/index.js

```diff
@@ -75,13 +63,28 @@ export default compose( [
 		return {
 			hasMultiSelection: hasMultiSelection(),
 			selectedBlockClientIds,
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		const { getBlocksByClientId } = select( 'core/editor' );
+		const { removeBlocks } = dispatch( 'core/editor' );
+
+		return {
+			getSelectedBlocks: function() {
```

Comment:
Maybe I've missed something, but I don't see why it should be `withDispatch` creating this function, instead of `withSelect`.

Reply:
Or, I could see it being an `onCopy` prop passed by `withDispatch`. Essentially, I'd see `withDispatch` only as providing some action handler, not a getter.

Reply:
`onCopy` is passed from `withDispatch` as of 59ec0a1.

> > Should we go further with that refactor and move both event handlers here?
> I'm not totally following what's referred to by "both".

We could do the same for `onCut`. Should we? I'm not sure whether referencing `onCopy` would be an issue, but I assume it will work.

Reply:
It might be very difficult to achieve if we follow with your other suggestion in https://github.com/WordPress/gutenberg/pull/11851#discussion_r236743156.